### PR TITLE
[Monitor][Ingestion] Add troubleshooting guide

### PR DESCRIPTION
**Pull request by pvaneck**
This adds a TROUBLESHOOTING.md file for Monitor Ingestion which gives users come common problem resolution guidance. 

Closes: https://github.com/Azure/azure-sdk-for-python/issues/29073

### CODE REVIEW

**Comment by pvaneck** [comment]
@scottaddie, This should be pretty consistent with https://github.com/Azure/azure-sdk-for-java/pull/34944 (after your comments).  Ran the troubleshooting guide through Acrolinx and it seems happy (minus what I think is a false positive, telling me to use "have" when "has" makes sense:  "Check that the application or user making the request has sufficient permissions")

File: sdk/monitor/azure-monitor-ingestion/TROUBLESHOOTING.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+### Troubleshooting slow logs upload
+
+If you experience delays when uploading logs, it could be due to reaching service limits, which may trigger the rate limiter to throttle your client. To determine if your client has reached service limits, you can enable logging and check if the service is returning errors with an HTTP status code 429. For more information on service limits, see the [Azure Monitor service limits documentation][ingestion_service_limits].
+
+To enable client logging and to troubleshoot this issue further, see the [Enable client logging](#enable-client-logging) section.
+
+## Extra azure-core configurations
+
+Some properties including `retry_mode`, `timeout`, `connection_verify` can be configured by passing them in as keyword arguments to the SDK method calls. See [configurations][azure_core_config] for a list of all such properties.
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/TROUBLESHOOTING.md:105]:
```suggestion
Some properties, including `retry_mode`, `timeout`, and `connection_verify`, can be configured by passing them in as keyword arguments to the library method calls. See [configurations][azure_core_config] for a list of all such properties.
```

File: sdk/monitor/azure-monitor-ingestion/TROUBLESHOOTING.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+1. If the user or application is granted sufficient privileges to upload logs, ensure you're authenticating as that user/application. If you're authenticating using the [DefaultAzureCredential](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/identity/azure-identity/README.md#authenticating-with-defaultazurecredential), check the logs to verify that the credential used is the one you expected. To enable logging, see the [Enable client logging](#enable-client-logging) section.
+1. The permissions may take up to 30 minutes to propagate. So, if the permissions were granted recently, retry after some time.
+
+### Troubleshooting missing logs
+
+When you send logs to Azure Monitor for ingestion, the request may succeed, but you may not see the data appearing in the designated Log Analytics workspace table as configured in the DCR. To investigate and resolve this issue, ensure the following items:
+
+* Double-check that you're using the correct data collection endpoint when creating the `LogsIngestionClient`. Using the wrong endpoint can result in data not being properly sent to the Log Analytics workspace.
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/TROUBLESHOOTING.md:87]:
```suggestion
* Double-check that you're using the correct DCE when creating the `LogsIngestionClient`. Using the wrong endpoint can result in data not being properly sent to the Log Analytics workspace.
```

File: sdk/monitor/azure-monitor-ingestion/TROUBLESHOOTING.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    * [Troubleshooting slow logs upload](#troubleshooting-slow-logs-upload)
+* [Extra azure-core configurations](#extra-azure-core-configurations)
+
+## General troubleshooting
+
+Monitor Ingestion raises exceptions defined in [Azure Core][azure_core_exceptions].
+
+### Enable client logging
+
+To troubleshoot issues with Azure Monitor Ingestion library, it's important to first enable logging to monitor the behavior of the application. The errors and warnings in the logs generally provide useful insights into what went wrong and sometimes include corrective actions to fix issues.
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/TROUBLESHOOTING.md:23]:
```suggestion
To troubleshoot issues with the Azure Monitor Ingestion library, it's important to first enable logging to monitor the behavior of the application. The errors and warnings in the logs generally provide useful insights into what went wrong and sometimes include corrective actions to fix issues.
```

File: sdk/monitor/azure-monitor-ingestion/TROUBLESHOOTING.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+This library uses the standard [logging][python_logging] library for logging. Basic information about HTTP sessions, such as URLs and headers, is logged at the INFO level. Detailed `DEBUG` level logging, including request/response bodies and unredacted headers, can be enabled on a client with the `logging_enable` argument:
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/TROUBLESHOOTING.md:25]:
```suggestion
This library uses the standard [logging][python_logging] library for logging. Basic information about HTTP sessions, such as URLs and headers, is logged at the `INFO` level. Detailed `DEBUG` level logging, including request/response bodies and unredacted headers, can be enabled on a client with the `logging_enable` argument:
```

File: sdk/monitor/azure-monitor-ingestion/TROUBLESHOOTING.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+```text
+Upload failed: (OperationFailed) The authentication token provided does not have access to ingest data for the data collection rule with immutable Id...
+```
+
+To resolve this issue:
+
+1. Check that the application or user making the request has sufficient permissions:
+   * See this document to [manage access to data collection rule][dcr_role_permissions].
+   * To ingest logs, ensure the user or service principal is assigned the **Monitoring Metrics Publisher** role for the data collection rule.
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/TROUBLESHOOTING.md:79]:
```suggestion
   * To ingest logs, ensure the user or service principal is assigned the **Monitoring Metrics Publisher** role for the DCR.
```

**Review by scottaddie — CHANGES_REQUESTED** [review]

File: sdk/monitor/azure-monitor-ingestion/TROUBLESHOOTING.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    * [Troubleshooting running async APIs](#errors-with-running-async-apis)
+* [Troubleshooting logs ingestion](#troubleshooting-logs-ingestion)
+    * [Troubleshooting authorization errors](#troubleshooting-authorization-errors)
+    * [Troubleshooting missing logs](#troubleshooting-missing-logs)
+    * [Troubleshooting slow logs upload](#troubleshooting-slow-logs-upload)
+* [Extra azure-core configurations](#extra-azure-core-configurations)
+
+## General troubleshooting
+
+Monitor Ingestion raises exceptions defined in [Azure Core][azure_core_exceptions].
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/TROUBLESHOOTING.md:19]:
```suggestion
The Monitor Ingestion library raises exceptions defined in [Azure Core][azure_core_exceptions].
```

File: sdk/monitor/azure-monitor-ingestion/TROUBLESHOOTING.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+client = LogsIngestionClient(endpoint, credential, logging_enable=True)
+```
+
+Similarly, `logging_enable` can enable detailed logging for a single operation, even when it isn't enabled for the client:
+
+```python
+client.upload(rule_id, stream_name, logs, logging_enable=True)
+```
+
+See full Python SDK logging documentation with examples [here][sdk_logging_docs].
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/TROUBLESHOOTING.md:49]:
```suggestion
For logging documentation with examples, see [Configure logging in the Azure libraries for Python][sdk_logging_docs].
```

**Review by scottaddie — APPROVED** [review]

**Pull request merged**